android/envoy/src/main/kotlin/org/greatfire/envoy/EnvoyTest.kt

```diff
@@ -89,196 +100,213 @@ data class EnvoyTest(
+            } else {
+                Log.d(TAG, "Cronet failed")
+            }
+            requestDone.open()
+        }
 
-                // XXX set DOH server for Go code here?
-                // or keep it in connect()?
+        override fun onFailed(
+            request: UrlRequest?,
+            info: UrlResponseInfo?,
+            error: CronetException?,
+        ) {
+            Log.d(TAG, "Cronet failed: " + error)
+            requestDone.open()
+        }
 
-                hostname?.let {
-                    val echConfigList = state.dns.getECHConfig(hostname)
-                    state.iep?.let {
-                        it.setEnvoyUrl(url, echConfigList)
+        override fun onCanceled(
+            request: UrlRequest?,
+            info: UrlResponseInfo?,
+        ) {
+            Log.d(TAG, "Cronet canceled.")
+            requestDone.open()
+        }
+
+        // Wait until cronet is done
+        @Throws(IOException::class)
+        fun blockForResponse() {
+            requestDone.block()
+        }
+    }
+
+    suspend fun testStandardProxy(proxyUri: Uri, testResponseCode: Int): Boolean {
+        Log.d(TAG, "Testing standard proxy $proxyUri")
 
-                        Log.d(TAG, "Starting ECH with $url $echConfigList")
+        if (proxyUri.host.isNullOrEmpty()) {
+            Log.e(TAG, "Empty proxy host!?")
+            return false
+        }
 
-                        // this uses the Go version of IsItUpYet before return
-                        it.start(IEnvoyProxy.EnvoyEch, "")
-                        val addr = it.localAddress(IEnvoyProxy.EnvoyEch)
-                        return addr
-                    }
-                }
-                return ""
+        // only socks5 and http are supported here
+        var proxyType = when(proxyUri.scheme) {
+            "socks5" -> Proxy.Type.SOCKS
+            "http" -> Proxy.Type.HTTP
+            else -> {
+                Log.e(TAG, "Unsupported proxy scheme")
+                return false
             }
-            EnvoyServiceType.HYSTERIA2 -> {
-                state.iep?.let {
-                    it.hysteria2Server = url
-                    it.start(IEnvoyProxy.Hysteria2, "")
-                    val addr = it.localAddress(IEnvoyProxy.Hysteria2)
-                    EnvoyConnectionTests.isItUpYet(addr)
-                    return addr
-                }
-                return ""
+        }
+
+        // apparently knowing the default port for the protocol is too
+        // much for the Uri libaray
+        var port = proxyUri.port
+        if (port == -1) {
+            port = when(proxyUri.scheme) {
+                "http"   -> 80
+                "socks5" -> 1080
+                else  -> 80 // ? error?
             }
-            EnvoyServiceType.SHADOWSOCKS -> {
-                // sadly this new code doesn't work, see the comments there
-                //
-                // shadowsocks = EnvoyShadowsocks(url, state.ctx!!)
```

Comment:
are we going to toss out my new Shadowsocks code? If so, we should remove the EnvoyShadowsocks file as well

Reply:
sorry, i lost track of the status of this.  is it something you're still working on or will it not work?